configure.ac

```diff
@@ -351,70 +351,26 @@ dnl options for contributed samplers
 
-dnl Jansson support
-AC_ARG_WITH([jansson],
-        [AS_HELP_STRING([--with-jansson],
-                [build against Jansson library @<:@default=check@:>@])],
-                [AS_IF([test "x$withval" != xyes -a "x$withval" != xno],
-                        [JANSSON_CFLAGS="-I$withval/include"
-                         JANSSON_LDFLAGS="-Wl,-rpath -Wl,$withval/lib -L$withval/lib"]
-                )],
-        [with_jansson=check])
-AC_SUBST([JANSSON_CFLAGS])
-AC_SUBST([JANSSON_LDFLAGS])
-have_jansson=yes
-AS_IF([test "x$with_jansson" != xno], [
-        save_LDFLAGS=$LDFLAGS
-        LDFLAGS="$LDFLAGS $JANSSON_LDFLAGS"
-        save_CFLAGS=$CFLAGS
-        CFLAGS="$CFLAGS $JANSSON_CFLAGS"
-        save_CPPFLAGS=$CPPFLAGS
-        CPPFLAGS="$CPPFLAGS $JANSSON_CFLAGS"
-        AC_CHECK_HEADER(jansson.h, [],[have_jansson=no])
-        AC_CHECK_LIB(jansson, json_real_value, [],[have_jansson=no])
-        CPPFLAGS=$save_CPPFLAGS
-        CFLAGS=$save_CFLAGS
-        LDFLAGS=$save_LDFLAGS
-])
-
 dnl Variorum dependency support
-AC_MSG_CHECKING([for Variorum library])
-AC_ARG_WITH([variorum],
-        [AS_HELP_STRING([--with-variorum],
-                [build against Variorum library @<:@default=check@:>@])],
-                [AS_IF([test "x$withval" != xyes -a "x$withval" != xno],
-                        [VARIORUM_CFLAGS="-I$withval/include"
-                         VARIORUM_LDFLAGS="-Wl,-rpath -Wl,$withval/lib -L$withval/lib"]
-                )],
-        [with_variorum=check])
-have_variorum=yes
-AC_SUBST([VARIORUM_CFLAGS])
-AC_SUBST([VARIORUM_LDFLAGS])
-AS_IF([test "x$with_variorum" != xno], [
-        save_LDFLAGS=$LDFLAGS
-        LDFLAGS="$LDFLAGS $VARIORUM_LDFLAGS"
-        save_CFLAGS=$CFLAGS
-        CFLAGS="$CFLAGS $VARIORUM_CFLAGS"
-        save_CPPFLAGS=$CPPFLAGS
-        CPPFLAGS="$CPPFLAGS $VARIORUM_CFLAGS"
-        AC_CHECK_HEADER(variorum.h,[],[have_variorum=no])
-        AC_CHECK_HEADER(variorum_topology.h, [],[have_variorum=no])
-        AC_CHECK_LIB(variorum, variorum_get_node_power_json, [],[have_variorum=no])
-        CPPFLAGS=$save_CPPFLAGS
-        CFLAGS=$save_CFLAGS
-        LDFLAGS=$save_LDFLAGS
-])
-AS_IF([test "x$have_variorum" = xno], [
-       AS_IF([test "x$with_variorum" = xcheck], [
-              AC_MSG_WARN("libvariorum not found, building without variorum")],
-              [AC_MSG_ERROR("libvariorum not found")
-       ])
-])
-AM_CONDITIONAL([ENABLE_VARIORUM_SAMPLER], [test x$have_variorum = xyes -a x$have_jansson = xyes])
-AM_COND_IF([ENABLE_VARIORUM_SAMPLER],[
-        AC_DEFINE([ENABLE_VARIORUM_SAMPLER], [], [Build Variorum sampler])
+AC_ARG_ENABLE([variorum],
+    [AS_HELP_STRING([--enable-variorum], [require components that depend upon libvariorum (and libjansson) @<:@default=check@:>@])],
+    [],
+    [enable_variorum="check"])
+AS_IF([test "x$enable_variorum" != xno],[
+    AC_LIB_HAVE_LINKFLAGS([variorum], [], [#include <variorum.h>])
```

Comment:
There should only be one call to AC_LIB_HAVE_LINKFLAGS for each library. If you want to have multiple include lines in the test, I believe that you can make the [includes] options multiline and have multiple #include statements.